### PR TITLE
refactor: add proto enum for ConsumerPhase

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: proto/interchain_security/ccv/provider/v1/provider.proto [hunk 1]
```diff
@@ -464,3 +464,20 @@ message PowerShapingParameters {
 // ConsumerIds contains consumer ids of chains
 // Used so we can easily (de)serialize slices of strings
 message ConsumerIds { repeated string ids = 1; }
+
+// ConsumerPhase indicates the phases of a consumer chain according to ADR 018
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/provider.proto:468]:
```suggestion
// ConsumerPhase indicates the phases of a consumer chain according to ADR 019
```

File: proto/interchain_security/ccv/provider/v1/provider.proto [hunk 2]
```diff
@@ -464,3 +464,20 @@ message PowerShapingParameters {
...
+enum ConsumerPhase {
+  // UNSPECIFIED defines an empty phase.
+  CONSUMER_PHASE_UNSPECIFIED = 0;
+  // REGISTERED defines the phase in which a consumer chain has been assigned a unique consumer id.
+  // A chain in this phase cannot yet launch.
+  CONSUMER_PHASE_REGISTERED = 1;
+  // INITIALIZED defines the phase in which a consumer chain has set all the needed parameters to launch but
+	// has not yet launched (e.g., because the `spawnTime` of the consumer chain has not yet been reached).
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/provider.proto:476]: nit: Indentation here  and below.

File: proto/interchain_security/ccv/provider/v1/provider.proto [hunk 3]
```diff
@@ -464,3 +464,20 @@ message PowerShapingParameters {
 // ConsumerIds contains consumer ids of chains
 // Used so we can easily (de)serialize slices of strings
 message ConsumerIds { repeated string ids = 1; }
+
+// ConsumerPhase indicates the phases of a consumer chain according to ADR 018
+enum ConsumerPhase {
+  // UNSPECIFIED defines an empty phase.
+  CONSUMER_PHASE_UNSPECIFIED = 0;
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/provider.proto:471]: Why do we repeat `CONSUMER_PHASE` in here and then everything ends up being `ConsumerPhase_CONSUMER_PHASE_INITIALIZED` having consumer phase twice? Why not call this `UNSPECIFIED`, `INITIALIZED`, etc.?

File: x/ccv/provider/keeper/key_assignment.go [hunk 4]
```diff
@@ -403,12 +403,14 @@ func (k Keeper) AssignConsumerKey(
 	validator stakingtypes.Validator,
 	consumerKey tmprotocrypto.PublicKey,
 ) error {
-	phase, found := k.GetConsumerPhase(ctx, consumerId)
-	if !found || phase == Stopped {
-		//check that the consumer chain is either registered, initialized, or launched
+	phase := k.GetConsumerPhase(ctx, consumerId)
+	if phase != types.ConsumerPhase_CONSUMER_PHASE_REGISTERED &&
```
Comment (insumity) [inline x/ccv/provider/keeper/key_assignment.go:407]: Nice!

File: proto/interchain_security/ccv/provider/v1/provider.proto (hunk 3 above)
Comment (mpoke) [reply proto/interchain_security/ccv/provider/v1/provider.proto:471]: This is how is done in the SDK https://github.com/cosmos/cosmos-sdk/blob/355f4d7e422f111b6a61243872adc6d5661f4cad/x/gov/proto/cosmos/gov/v1/gov.proto#L17

**Review by insumity — APPROVED** [review]
Besides [this](https://github.com/cosmos/interchain-security/pull/2182/files#r1734605365) comment, left some other minor ones.

LGTM

**Pull request merged**